app/controllers/AmendControllerCacheHelper.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import common.Strings
+import models.ProtectionModel
+import play.api.mvc.Results.Redirect
+import services.SessionCacheService
+import models.amendModels._
+import play.api.mvc.{MessagesRequest, Result}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait AmendControllerCacheHelper {
+
+  val sessionCacheService: SessionCacheService
+
+  def fetchAmendProtectionModel(cacheKey: String)(implicit request: MessagesRequest[_]): Future[Option[AmendProtectionModel]] = {
+    sessionCacheService.fetchAndGetFormData[AmendProtectionModel](cacheKey)
+  }
+
+  def saveAndRedirectToSummary(cacheKey: String, amendModel: AmendProtectionModel)(implicit request: MessagesRequest[_], ec: ExecutionContext): Future[Result] = {
+    sessionCacheService
+      .saveFormData[AmendProtectionModel](cacheKey, amendModel)
+      .map { _ =>
+        val updatedProtectionType = Strings.protectionTypeString(amendModel.updatedProtection.protectionType).toLowerCase
+        val updatedStatus = Strings.statusString(amendModel.updatedProtection.status).toLowerCase
+
+        Redirect(routes.AmendsController.amendsSummary(updatedProtectionType, updatedStatus))
```

Comment:
I'd suggest to remove redirection from this method. The class is called `CacheHelper` and I'd only put cache-related operations here.

Reply:
I agree that if all these controllers redirect to the same page, it would be handy to extract it to a common method. But I don't like the idea of doing cache update AND redirection in a single method. It breaks single responsibility principle.

Maybe you could extract redirection to a new method and call it from controllers after cache is updated? You could have the method in this trait, but then I'd encourage a name change.